app/models/weam.rb

```diff
@@ -42,18 +42,16 @@ class Weam < ActiveRecord::Base
   validates :institution, presence: true
   validates :bah, numericality: true, allow_blank: true
 
-  validate :validate_derived_fields
+  validate :derive_fields
 
   # Computes all fields that are dependent on other fields. Called in validation because
   # activerecord-import does not engage callbacks when saving
-  def validate_derived_fields
+  def derive_fields
```

Comment:
Could be mistaken, but it looks like you can enable callback calling. https://github.com/zdennis/activerecord-import/wiki/Callbacks

Either way, I do think it would be best to have a separate method for mutating those dependent attributes.

Outside the scope of this PR, but I do wonder if we could eliminate some of these dependent attributes–at least the ones that are equivalent to one other attribute. Similar things are happening in the frontend that are probably best simplified in the backend first.